grpc/py-cubic/cubic/client.py

```diff
@@ -0,0 +1,142 @@
+                                when connecting to an instance of cubic server that is 
+                                using a self-signed certificate.
+
+            clientCertificate:  PEM certificate as bytes string presented by this Client when
+                                connecting to a server. Use this when setting up mutually 
+                                authenticated TLS. The clientKey must also be provided.
+
+            clientKey:  PEM key as byte string presented by this Client when
+                        connecting to a server. Use this when setting up mutually 
+                        authenticated TLS. The clientKey must also be provided.
+
+            bufferSize: Bytes of audio sent to cubic server in each chunk if using streaming
+                        recognition. Default is 8192 bytes.
+        """
+
+        self.serverAddress = serverAddress
+        self.secure = secure
+        self.bufferSize = bufferSize
+        
+        if self.bufferSize <= 0:
+            raise ValueError('buffer size must be greater than 0')
+
+        if secure:
+            # using a TLS endpoint with optional certificates for mutual authentication
+            if clientCertificate is not None and clientKey is None:
+                raise ValueError("client key has not been provided")
+            if clientKey is not None and clientCertificate is None:
+                raise ValueError("client certificate has not been provided")
+            self._creds = grpc.ssl_channel_credentials(
+                root_certificates=serverCertificate, 
+                private_key=clientKey,
+                certificate_chain=clientCertificate)
+            self._channel = grpc.secure_channel(serverAddress, self._creds)
+        else:
+            self._channel = grpc.insecure_channel(serverAddress)
+
+        self._client = CubicStub(self._channel)
+
+    def __del__(self):
+        """ Closes and cleans up after Client. """
+        try:
+            self._channel.close()
+        except AttributeError:
+            # client wasn't 
+            pass
+
+    def Version(self):
+        """ Queries the server for its version. """
+        return self._client.Version(Empty())
+
+    def ListModels(self):
+        """ Retrieves a list of available speech recognition models. """
+        return self._client.ListModels(ListModelsRequest())
+        
+    def Recognize(self, cfg, audio):
+        """ Performs synchronous speech recognition: receive results after all audio
+        has been sent and processed. It is expected that this request be typically
+        used for short audio content: less than a minute long.  For longer content,
+        the `StreamingRecognize` method should be preferred.
+
+        Args:
+            cfg: RecognitionConfig object containing the model ID, timeout, audio encoding etc.
+
+            audio: a binary stream of data to send to cubic. 
```

Comment:
probably clarify that this is some object that implements the read() method. Otherwise clients might attempt to use a bytestring/etc.

Reply:
will do; should we also document each function in the actual sdk documentation instead of only the code files ?

Reply:
Maybe later.